packages/iov-core/src/jsrpcsigningserver.ts

```diff
@@ -112,41 +108,38 @@ export class JsonRpcSigningServer {
   }
 
   /**
-   * Handles a checked JsonRpcRequest
+   * Handles a checked JsRpcRequest
    *
-   * 1. convert JsonRpcRequest into calls to SigningServerCore
+   * 1. convert JsRpcRequest into calls to SigningServerCore
    * 2. call SigningServerCore
-   * 3. convert result to JSON-RPC format
+   * 3. convert result to JS RPC format
    */
-  public async handleChecked(request: JsonRpcRequest): Promise<JsonRpcResponse> {
+  public async handleChecked(request: JsRpcRequest): Promise<JsRpcResponse> {
     let call: RpcCall;
     try {
       call = parseRpcCall(request);
     } catch (error) {
-      const errorResponse: JsonRpcErrorResponse = {
-        jsonrpc: "2.0",
+      const errorResponse: JsRpcErrorResponse = {
         id: request.id,
         error: {
-          code: jsonRpcCodeMethodNotFound,
+          code: jsRpcCode.methodNotFound,
           message: error.toString(),
         },
       };
       return errorResponse;
     }
 
     try {
-      let response: JsonRpcSuccessResponse;
+      let response: JsRpcSuccessResponse;
       switch (call.name) {
         case "getIdentities":
           response = {
-            jsonrpc: "2.0",
             id: request.id,
             result: await this.core.getIdentities(call.reason, call.chainIds),
```

Comment:
I need to dig deeper in the code, but since JsRpcRequest.params can be Array or Dictionary, does parseRpcCall handle the array format properly, as you refer to x.reason below

Reply:
`parseRpcCall` ensures the dictionary parameters are used and checks the type of those fields